File: python/vedant_bhatnagar/DevsnestProblems_THA/Phase_2/Trees/SameTree.py
def solve(p, q):
    # CODE HERE
    if (p and not q) or (q and not p):
        return 0
    if not p and not q:
        return 1
    else:
        return 1 if p.val == q.val and solve(p.left,q.left) and solve(p.right,q.right) else 0

File: java/vansh_jain/trees/SameTree.java
static int solve(TreeNode p, TreeNode q){
    //CODE HERE 
    if(p==null && q==null)
        return 1;
    if((p!=null && q==null) || (p==null && q!=null) || (p!=null && q!=null && p.val!=q.val))
        return 0;
    
    // System.out.println(p.val+" "+q.val);
    if(solve(p.left,q.left)==1 &&
    solve(p.right,q.right) ==1)
        return 1;
    return 0;
    }